File: java/runner/src/main/java/com/example/evernote/LocalStore.java
package com.example.evernote;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;

public class LocalStore {

    private final Path base;

    private final Path enex_batch;
    private final Path enex_batch_csv;
    private final Path enex_batch_exports;
    private final Path enex_batch_parsed;

    private final Path internet_single;
    private final Path internet_single_notes;
    private final Path internet_single_oauth;
    private final Path internet_single_tagged;
    private final Path internet_single_csv;
    private final Path internet_single_training;

    private final Path images_tmp;

    private final Path mappings;

    private static final LocalStore singleton;

    static {
        LocalStore ls = null;
        try {
            ls = new LocalStore();
        } catch (Exception e) {
            e.printStackTrace(System.out);
        }
        singleton = ls;
    }

    private LocalStore() throws IOException {

        //test
        //base = create("C:\\Users\\Jonas\\Downloads\\0", ".jg-evernote");

        base = create(Paths.get(System.getProperty("user.home")).toString(), ".jg-evernote");

        enex_batch = create(base.toString(), "enex-batch");
        enex_batch_csv = create(enex_batch.toString(), "csv");
        enex_batch_exports = create(enex_batch.toString(), "exports");
        enex_batch_parsed = create(enex_batch.toString(), "parsed");

        internet_single = create(base.toString(), "internet-single");
        internet_single_notes = create(internet_single.toString(), "notes");
        internet_single_oauth = create(internet_single.toString(), "oauth");
        internet_single_tagged = create(internet_single.toString(), "tagged");
        internet_single_csv = create(internet_single.toString(), "csv");
        internet_single_training= create(internet_single.toString(), "training");

        images_tmp = create(base.toString(), "images-tmp");

        mappings = create(base.toString(), "mappings");
    }

    private Path create(String b, String s) throws IOException {
        Path path = Paths.get(b, s);
        return Files.createDirectories(path);
    }

    public static LocalStore getSingleton() {
        return singleton;
    }

    public Path getEnex_batch_csv() {
        return enex_batch_csv;
    }

    public Path getEnex_batch_exports() {
        return enex_batch_exports;
    }

    public Path getEnex_batch_parsed() {
        return enex_batch_parsed;
    }

    public Path getInternet_single_notes() {
        return internet_single_notes;
    }

    public Path getInternet_single_oauth() {
        return internet_single_oauth;
    }

    public Path getInternet_single_tagged() { return internet_single_tagged; }

    public Path getInternet_single_csv() { return internet_single_csv; }

    public Path getInternet_single_training() { return internet_single_training; }

    public Path getImages_tmp() { return images_tmp; }

    public Path getMappings() { return mappings; }

}

File: python/src/local_store.py
import os
from pathlib import Path

PATHS: dict[str, str] = {}

BASE = "base"

MODEL_ARTIFACTS = "model_artifacts"

IMAGES_TMP = "images_tmp"

MAPPINGS = "mappings"

ENEX_BATCH = "enex_batch"
ENEX_BATCH_CSV = "enex_batch_csv"
ENEX_BATCH_EXPORTS = "enex_batch_exports"
ENEX_BATCH_PARSED = "enex_batch_parsed"

INTERNET_SINGLE = "internet_single"
INTERNET_SINGLE_NOTES = "internet_single_notes"
INTERNET_SINGLE_OAUTH = "internet_single_oauth"
INTERNET_SINGLE_TAGGED = "internet_single_tagged"
INTERNET_SINGLE_CSV = "internet_single_csv"
INTERNET_SINGLE_TRAINING = "internet_single_training"

PATHS[BASE] = os.path.join(os.getenv("DATA", Path.home()), ".jg-evernote")

PATHS[MODEL_ARTIFACTS] = os.path.join(PATHS[BASE], "model-artifacts")

PATHS[ENEX_BATCH] = os.path.join(PATHS[BASE], "enex-batch")
PATHS[ENEX_BATCH_CSV] = os.path.join(PATHS[ENEX_BATCH], "csv")
PATHS[ENEX_BATCH_EXPORTS] = os.path.join(PATHS[ENEX_BATCH], "exports")
PATHS[ENEX_BATCH_PARSED] = os.path.join(PATHS[ENEX_BATCH], "parsed")

PATHS[INTERNET_SINGLE] = os.path.join(PATHS[BASE], "internet-single")
PATHS[INTERNET_SINGLE_NOTES] = os.path.join(PATHS[INTERNET_SINGLE], "notes")
PATHS[INTERNET_SINGLE_OAUTH] = os.path.join(PATHS[INTERNET_SINGLE], "exports")
PATHS[INTERNET_SINGLE_TAGGED] = os.path.join(PATHS[INTERNET_SINGLE], "tagged")
PATHS[INTERNET_SINGLE_CSV] = os.path.join(PATHS[INTERNET_SINGLE], "csv")
PATHS[INTERNET_SINGLE_TRAINING] = os.path.join(PATHS[INTERNET_SINGLE], "training")

PATHS[MAPPINGS] = os.path.join(PATHS[BASE], "mappings")

PATHS[IMAGES_TMP] = os.path.join(PATHS[BASE], "images-tmp")

for key, path in PATHS.items():
    Path(path).mkdir(parents=True, exist_ok=True)

print(f"Ensured paths for local store '{PATHS[BASE]}'.")